ingress/controllers/nginx/README.md

```diff
@@ -186,60 +187,363 @@ data:
 Please check the [udp services](examples/udp/README.md) example
 
 
-## Custom NGINX configuration
+### Annotations
 
-Using a ConfigMap it is possible to customize the defaults in nginx.
+|Name                 |type|
+|---------------------------|------|
+|[ingress.kubernetes.io/rewrite-target](#rewrite)|URI|
+|[ingress.kubernetes.io/add-base-url](#rewrite)|true or false|
+|[ingress.kubernetes.io/limit-connections](#rate-limiting)|number|
+|[ingress.kubernetes.io/limit-rps](#rate-limiting)|number|
+|[ingress.kubernetes.io/auth-type](#authentication)|basic or digest|
+|[ingress.kubernetes.io/auth-secret](#authentication)|string|
+|[ingress.kubernetes.io/auth-realm](#authentication)|string|
+|[ingress.kubernetes.io/ssl-redirect](#server-side-https-enforcement-through-redirect)|true or false|
+|[ingress.kubernetes.io/upstream-max-fails](#custom-nginx-upstream-checks)|number|
+|[ingress.kubernetes.io/upstream-fail-timeout](#custom-nginx-upstream-checks)|number|
+|[ingress.kubernetes.io/secure-upstream](#secure-nginx-upstreams)|true or false|
+|[ingress.kubernetes.io/whitelist-source-range](#whitelist-source-range)|CIDR|
 
-Please check the [tcp services](examples/custom-configuration/README.md) example
+### Custom NGINX upstream checks
```

Comment:
I would just call this health checks, instead of using "upstream" in the heading